src/main/java/org/opensearch/neuralsearch/query/HybridQuery.java

```diff
@@ -77,20 +76,20 @@ public String toString(String field) {
     /**
      * Re-writes queries into primitive queries. Callers are expected to call rewrite multiple times if necessary,
      * until the rewritten query is the same as the original query.
-     * @param reader
+     * @param indexSearcher
      * @return
      * @throws IOException
      */
     @Override
-    public Query rewrite(IndexReader reader) throws IOException {
+    public Query rewrite(IndexSearcher indexSearcher) throws IOException {
```

Comment:
Why do we need to switch? Isnt this an overriden method - what happens to the other method if called?

Reply:
Lucene has [some logic](https://github.com/apache/lucene/blob/branch_9_10/lucene/core/src/java/org/apache/lucene/search/Query.java#L55-L60) and [different code paths](https://github.com/apache/lucene/blob/branch_9_10/lucene/core/src/java/org/apache/lucene/search/Query.java#L98-L100) depending on what exact method has been overriden. With today's code (where we override `rewrite(IndexReader)`) in certain scenarios doc iterator may have empty results (no doc ids). In particular I noticed that for Term query in scenarios when there are multiple term sub-queries and more than one query has matches. I suspect that happens because of rewrite. We'd stop doing rewrites too early on a first sub-query with matched docs, while we should continue rewrite cycles for other term sub-queries.